sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Listeners/ServiceBusListener.cs

```diff
@@ -123,27 +123,35 @@ public async Task StartAsync(CancellationToken cancellationToken)
             {
                 throw new InvalidOperationException("The listener has already been started.");
             }
+            Started = true;
 
-            if (_singleDispatch)
+            try
             {
-                if (_isSessionsEnabled)
+                if (_singleDispatch)
                 {
-                    _sessionMessageProcessor.Value.Processor.ProcessMessageAsync += ProcessSessionMessageAsync;
-                    await _sessionMessageProcessor.Value.Processor.StartProcessingAsync(cancellationToken).ConfigureAwait(false);
+                    if (_isSessionsEnabled)
+                    {
+                        _sessionMessageProcessor.Value.Processor.ProcessMessageAsync += ProcessSessionMessageAsync;
+                        await _sessionMessageProcessor.Value.Processor.StartProcessingAsync(cancellationToken).ConfigureAwait(false);
+                    }
+                    else
+                    {
+                        _messageProcessor.Value.Processor.ProcessMessageAsync += ProcessMessageAsync;
+                        await _messageProcessor.Value.Processor.StartProcessingAsync(cancellationToken).ConfigureAwait(false);
+                    }
+
+                    _concurrencyUpdateManager?.Start();
                 }
                 else
                 {
-                    _messageProcessor.Value.Processor.ProcessMessageAsync += ProcessMessageAsync;
-                    await _messageProcessor.Value.Processor.StartProcessingAsync(cancellationToken).ConfigureAwait(false);
+                    _batchLoop = RunBatchReceiveLoopAsync(_cancellationTokenSource.Token);
                 }
-
-                _concurrencyUpdateManager?.Start();
             }
-            else
+            catch
```

Comment:
could we add a comment here about what exception we'd expect to catch here and some context about why?

Reply:
Sure, there isn't a particular exception we are expecting, but if we get some error when trying to start, we need to mark the listener as not started so that the WebJobs host can try to start the listener again.

Reply:
I will add this as a comment.